From 2af48eff2e91a5e7aa84ac37e9e87f7633a02fd0 Mon Sep 17 00:00:00 2001
From: Chenhui Zhao <chenhui.zhao@freescale.com>
Date: Thu, 13 Nov 2014 18:44:57 +0800
Subject: [PATCH 1198/1207] powerpc/rcpm: fix compile error with non-SMP mode

Add asm/smp.h, and only compile the cluster related code
when CPU hotplug is enabled.

Change-Id: I264ff297ca1ee29bf8c477b09bce98eeb27140ee
Signed-off-by: Chenhui Zhao <chenhui.zhao@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/23752
Reviewed-by: Yang Li <LeoLi@freescale.com>
Reviewed-by: Zhuoyu Zhang <Zhuoyu.Zhang@freescale.com>
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Matthew Weigel <Matthew.Weigel@freescale.com>
[Lu:Original patch taken from
QorIQ-SDK-V1.7-SOURCE-20141218-yocto.iso]
Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 arch/powerpc/sysdev/fsl_rcpm.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/powerpc/sysdev/fsl_rcpm.c b/arch/powerpc/sysdev/fsl_rcpm.c
index 6621346..d410b9a 100644
--- a/arch/powerpc/sysdev/fsl_rcpm.c
+++ b/arch/powerpc/sysdev/fsl_rcpm.c
@@ -20,6 +20,7 @@
 #include <asm/cputhreads.h>
 #include <asm/fsl_pm.h>
 #include <asm/machdep.h>
+#include <asm/smp.h>
 
 #include <platforms/85xx/smp.h>
 
@@ -242,6 +243,7 @@ static void rcpm_v2_cpu_exit_state(int cpu, int state)
 	}
 }
 
+#ifdef CONFIG_HOTPLUG_CPU
 static void rcpm_v2_cluster_enter_state(int cpu, int state)
 {
 	int hw_cpu;
@@ -278,6 +280,7 @@ static void rcpm_v2_cluster_exit_state(int cpu, int state)
 		pr_err("%s: Unknown cluster PM state (%d)\n", __func__, state);
 	}
 }
+#endif
 
 
 static int rcpm_v2_plat_enter_state(int state)
@@ -354,6 +357,7 @@ bool rcpm_v2_cpu_ready(unsigned int cpu, int state)
 		if (in_be32(&rcpm_v2_regs->pcph30sr) & mask)
 			ret = true;
 		break;
+#ifdef CONFIG_HOTPLUG_CPU
 	case E500_PM_PCL10:
 		/* PCL10 state is only supported on e6500 for now. */
 		mask = 1 << (hw_cpu / THREAD_IN_CLUSTER);
@@ -361,6 +365,7 @@ bool rcpm_v2_cpu_ready(unsigned int cpu, int state)
 		if (in_be32(&rcpm_v2_regs->clpcl10sr) & mask)
 			ret = true;
 		break;
+#endif
 	default:
 		pr_err("%s: Unknown platform PM state (%d)\n",
 				__func__, state);
@@ -386,8 +391,10 @@ static const struct fsl_pm_ops qoriq_rcpm_v2_ops = {
 	.irq_unmask = rcpm_v2_irq_unmask,
 	.cpu_enter_state = rcpm_v2_cpu_enter_state,
 	.cpu_exit_state = rcpm_v2_cpu_exit_state,
+#ifdef CONFIG_HOTPLUG_CPU
 	.cluster_enter_state = rcpm_v2_cluster_enter_state,
 	.cluster_exit_state = rcpm_v2_cluster_exit_state,
+#endif
 	.plat_enter_state = rcpm_v2_plat_enter_state,
 	.set_ip_power = rcpm_v2_set_ip_power,
 	.freeze_time_base = rcpm_v2_freeze_time_base,
-- 
2.0.2

